src/data/CONTRIBUTORS.ts

```diff
@@ -106,8 +111,14 @@ export default {
 		jobs: [JOBS.MONK],
 	},
 }
+export default CONTRIBUTORS
+
+export interface Role {
+	i18n_id: string
+	text: string
+}
 
-export const ROLES = {
+export const ROLES: Record<string, Role> = {
```

Comment:
Doing the same for maintainers would be more annoying. Could just leave that one as `string`, but how's this for a workaround:

```ts
function record<T> (): <O extends Record<string, T>>(o: O) => O {
  return x => x
}
```

![image](https://user-images.githubusercontent.com/73085/47412083-a0135f80-d7a5-11e8-952a-0310329ef04f.png)

Note that it has to return a function because it's the only way to reuse a generic parameter in another generic parameter while preserving type inference.

It actually even preserves the individual types of each value; the only requirement is that the value must be a subtype of `T`:

![image](https://user-images.githubusercontent.com/73085/47412174-e49efb00-d7a5-11e8-9aac-569a12fb7584.png)

![image](https://user-images.githubusercontent.com/73085/47412194-f97b8e80-d7a5-11e8-8234-6e0211a4281f.png)

Reply:
This version of it produces nicer types:

```ts
function record<T> (): <K extends string>(o: Record<K, T>) => {[k in K]: T} {
  return x => x
}
```

Change the `{[k in K]: T}` to `Record<K, T>` to change how the tooltip looks like (will show a union of strings instead of an object).